week02/GroupAnagrams.cpp

```diff
@@ -0,0 +1,31 @@
+//TC:O(NK), N = input vector size, K=max string length
+// MC: O(NK)
+class Solution {
+public:
+    string hash(vector<int>v){
+        string temp="";
+        for(int i: v) temp+=to_string(i)+"#";
+        
+        return temp;
+    }
+    
+    vector<vector<string>> groupAnagrams(vector<string>& strs) {
+		vector<vector<string>> ans;
+        unordered_map<string, vector<string>> anagram;
+		
+		for(auto str: strs){
+			vector<int>freq(26, 0);
+			
+			for(char ch: str) freq[ch-'a']++;
+			
+            string key = hash(freq);
+            anagram[key].push_back(str);
+		}
+			
+		for(auto [key, value]: anagram){
+			ans.emplace_back(value);
+		}
+
+		return ans;
+    }
+};
```

Comment:
Brother, your code is not readable at all. Please look after your indentations. Variable naming and function naming need improvement in so many places.

Your `hash` function is not doing anything great. It's just creating another string from the given vector by concatenating `n` times (each concatenating takes O(N) time roughly).

Kindly refer to the [editorial](https://docs.google.com/document/d/1VipVcGyqiUDvFWbJ_yivTJucBRw3ct7HGZGGx-_CRsk/edit) for an improved version.